app/src/main/java/com/spascoding/taskycourse/feature_agenda/presentation/detail_screen/components/Controls.kt

```diff
@@ -0,0 +1,52 @@
+package com.spascoding.taskycourse.feature_agenda.presentation.detail_screen.components
+
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Close
+import androidx.compose.material.icons.outlined.Edit
+import androidx.compose.material3.Button
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import com.spascoding.taskycourse.R
+import com.spascoding.taskycourse.feature_agenda.presentation.detail_screen.DetailEvent
+import com.spascoding.taskycourse.feature_agenda.presentation.detail_screen.DetailViewModelState
+import com.spascoding.taskycourse.feature_agenda.presentation.utils.DATE_FORMAT
+
+@Composable
+fun Controls(
+    navController: NavController,
```

Comment:
Passing the navController to composables also quickly breaks the preview. I'd recommend rather bubbling up nav events with `onEvent` and responding to them in the root composable.